src/SimSpace/Config.hs

```diff
@@ -28,15 +33,30 @@ newtype WhitelistFiles = WhitelistFiles { unWhitelistFiles :: [FilePath] }
 
 
 {- | Filter files in a git repository by considering the files we do want to format. Normalize the raw files. -}
-filterFiles :: FormatFiles -> WhitelistFiles -> IO [FilePath]
-filterFiles (FormatFiles rawFiles) (WhitelistFiles rawWhitelist) = do
+filterFiles :: FormatFiles -> WhitelistFiles -> Bool -> IO [FilePath]
+filterFiles (FormatFiles rawFiles) (WhitelistFiles rawWhitelist) allFiles = do
   files <- traverse (makeAbsolute >=> makeRelativeToCurrentDirectory) rawFiles
   whitelist <- traverse (makeAbsolute >=> makeRelativeToCurrentDirectory) rawWhitelist
+  hasGit <- doesDirectoryExist ".git"
   let matchesFiles fp = elem "." files || any (flip isPrefixOf fp) files
       matchesWhitelist fp = elem "." whitelist || any (flip isPrefixOf fp) whitelist
       isValid fp = isSuffixOf ".hs" fp && matchesFiles fp && not (matchesWhitelist fp)
-  filter isValid . lines <$> readCreateProcess (shell "git ls-files") ""
+      srcFiles =
+        if not hasGit || allFiles
+          then Set.toList . mconcat <$> traverse listFilesRecursive ["."]
+          else lines <$> readCreateProcess (shell "git ls-files") ""
+  filter isValid <$> srcFiles
 
+listFilesRecursive :: FilePath -> IO (Set FilePath)
+listFilesRecursive dir = do
```

Comment:
I don't think this excludes .stack-work. Can you filter out hidden directories?

Reply:
Done